### PR TITLE
API Review for Release 1.2

### PR DESCRIPTION
**Pull request by shaneutt**
API Review PR for 1.2 release

**SEE THE CHANGELOG PR FIRST:** https://github.com/kubernetes-sigs/gateway-api/pull/3327

This PR wont be merged directly, as its based against `release-1.1` to specifically highlight the API changes since then.

Once we're done reviewing we'll simply close this without merging and the PR history will be for posterity, a `release-1.2` branch will be pushed up based on `main` once we're ready to release (If we have any changes that come out of this PR review, they will be PR'd against `main` prior to release).

### CODE REVIEW

File: CHANGELOG/1.2-TEAM.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# v1.2 Release Team
+
+| Enhancement | Leads |
+| - | - |
+| Infrastructure Labels and Annotations | @keithmattix, @snorwin |
+| HTTPRoute Timeouts and Durations | @kflynn, @xtineskim |
+| BackendProtocol Support | @dprotaso |
+| Percentage-based request mirroring | @jakebennert |
+| Backend TLS Config improvments | @mkosieradzki, @LiorLieberman |
```
Comment (sayboras) [inline CHANGELOG/1.2-TEAM.md:9]: typo: improvments -> improvements

File: CHANGELOG/1.2-CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+This feature's status is **Extended**, meaning that it is _optional_ for
+implementations to support.
+
+As there is no feature name or conformance tests available for this feature
+as yet, please see your implementation's documentation to see if it is
+supported.
+
+Relevant PRs:
+* Implement of GEP-995 by @howardjohn in [#2985](https://github.com/kubernetes-sigs/gateway-api/pull/2985)
+* Docs: route rule name by @guicassolato in [3299](https://github.com/kubernetes-sigs/gateway-api/pull/3299)
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply CHANGELOG/1.2-CHANGELOG.md]: In retrospect I dropped the commit for the changelog, we should follow up over here instead:

https://github.com/kubernetes-sigs/gateway-api/pull/3327

Sorry for any confusion!

File: CHANGELOG/1.2-TEAM.md (hunk 1 above)
Comment (shaneutt) [reply CHANGELOG/1.2-TEAM.md]: In retrospect I dropped the commit for the changelog, we should follow up over here instead:

https://github.com/kubernetes-sigs/gateway-api/pull/3327

Sorry for any confusion!

**Comment by shaneutt** [comment]
Release is complete!

**Pull request closed without merging**